File: easy/problems/0500-keyboard-row/solution.ts
function findWords(words: string[]): string[] {
    const rows = [
        new Set('qwertyuiop'),
        new Set('asdfghjkl'),
        new Set('zxcvbnm')
    ];

    const result: string[] = [];
    for (const word of words) {
        const wordLower = word.toLowerCase();
        for (const row of rows) {
            if ([...wordLower].every(char => row.has(char))) {
                result.push(word);
                break;
            }
        }
    }

    return result;
}

File: easy/problems/0500-keyboard-row/solution.cpp
class Solution {
public:
    vector<string> findWords(vector<string>& words) {
        vector<set<char>> rows = {
            set<char>{'q', 'w', 'e', 'r', 't', 'y', 'u', 'i', 'o', 'p'},
            set<char>{'a', 's', 'd', 'f', 'g', 'h', 'j', 'k', 'l'},
            set<char>{'z', 'x', 'c', 'v', 'b', 'n', 'm'}
        };

        vector<string> result;
        for (const string& word : words) {
            for (const auto& row : rows) {
                bool found = true;
                for (char c : word) {
                    if (row.find(tolower(c)) == row.end()) {
                        found = false;
                        break;
                    }
                }
                if (found) {
                    result.push_back(word);
                    break;
                }
            }
        }

        return result;
    }
};
